### PR TITLE
CLN: assorted cleanups

### PR DESCRIPTION
**Pull request by jbrockmendel**
Mostly annotations

### CODE REVIEW

File: pandas/_libs/tslibs/timedeltas.pyx [hunk 1]
```diff
@@ -308,7 +313,7 @@ cdef inline int64_t cast_from_unit(object ts, object unit) except? -1:
         int64_t m
         int p
 
-    m, p = precision_from_unit(unit)
+    m, p = precision_from_unit(unit or "ns")
```
Comment (jreback) [inline pandas/_libs/tslibs/timedeltas.pyx:316]: is this hit? (the ns)

File: pandas/core/indexes/base.py [hunk 2]
```diff
@@ -5516,9 +5516,9 @@ def ensure_index(index_like, copy=False):
         # clean_index_list does the equivalent of copying
         # so only need to do this if not list instance
         if copy:
-            from copy import copy
+            from copy import copy as copy_func
```
Comment (jreback) [inline pandas/core/indexes/base.py:5519]: this should be at the top of the file

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/indexes/base.py (hunk 2 above)
Comment (jbrockmendel) [reply pandas/core/indexes/base.py]: updated

**Review by jreback — APPROVED** [review]

**Review by WillAyd — APPROVED** [review]

**Comment by WillAyd** [comment]
Thanks @jbrockmendel

**Pull request merged**